src/meta/src/hummock/manager/mod.rs

```diff
@@ -1942,6 +1943,58 @@ where
         &self.cluster_manager
     }
 
+    pub fn suggest_compactor_cores(&self, info: &ScaleCompactorInfo) -> u64 {
+        let mut suggest_core = info.scale_out_cores() + info.total_cores;
+        match self.compactor_manager.suggest_scale_policy() {
+            ScalePolicy::ScaleIn(core) => {
+                if core + info.running_cores < info.total_cores {
+                    suggest_core = info.total_cores - core;
+                } else {
+                    suggest_core = info.total_cores;
+                }
+            }
```

Comment:
`info.scale_out_cores() > 0` and `self.compactor_manager.suggest_scale_policy() == ScalePolicy::ScaleIn(core)` seems to be conflicting and happens when at least one compactor's CPU < 80% and the LSM is still not balanced. 

I think the reason for this is the task limit being too small for the CPU count, e.g. CPU usage < threshold even if task num hit the limit. So we should prioritize scale in only when `info.scale_out_cores() == 0 or < some small threshold`.

Reply:
> info.scale_out_cores() > 0 and self.compactor_manager.suggest_scale_policy() == ScalePolicy::ScaleIn(core)
yes, I think is a little confusing about suggest_core when lsm is still not balanced. Policy to suggest ScalePolicy only relies on the state of Compactor, it has no need to be concerned about `lsm shape`.I've modified the running_core calculation to try to avoid this. 

For the above case, all compactors need to run at a low load for more than 10 minutes and the lsm is constantly unbalanced, which I think is an unusual behavior and we should pay attention to it. So how about we prioritize adding logs for this case?

Reply:
> I've modified the running_core calculation to try to avoid this.

This part looks more reasonable now.

> For the above case, all compactors need to run at a low load for more than 10 minutes and the lsm is constantly unbalanced

The condition for downgrading compaction state is `idle_count > 0`, so I think it's "**at least one** compactor need to run at a low load for more than 10 minutes" 🤔 

> prioritize adding logs for this case

+1